qtawesome/iconic_font.py

```diff
@@ -197,14 +202,31 @@ def hook(obj):
         with open(os.path.join(directory, charmap_filename), 'r') as codes:
             self.charmap[prefix] = json.load(codes, object_hook=hook)
 
-        id_ = QFontDatabase.addApplicationFont(os.path.join(directory, ttf_filename))
+        md5_hashes = {'fontawesome-webfont.ttf':
+                        'a3de2170e4e9df77161ea5d3f31b2668',
+                    'elusiveicons-webfont.ttf':
+                        '207966b04c032d5b873fd595a211582e'}
+        ttf_hash = md5_hashes.get(ttf_filename, None)
+        if ttf_hash is not None:
+            hasher = hashlib.md5()
+            with open(os.path.join(directory, ttf_filename), 'rb') as tff_font:
+                buffer = tff_font.read()
+                hasher.update(buffer)
+            ttf_calculated_hash_code = hasher.hexdigest()
+            if ttf_calculated_hash_code != ttf_hash:
+                raise FontError(u"Font is corrupt at: '{0}'".format(
```

Comment:
Sorry, i forgot this message is not going to be displayed in Qt. So there's no need of this change.